cdap-security/src/main/java/co/cask/cdap/security/authorization/ForwardingAuthorizationContext.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authorization;
+
+import co.cask.cdap.api.Admin;
+import co.cask.cdap.api.Transactional;
+import co.cask.cdap.api.TxRunnable;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.data.DatasetInstantiationException;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetManagementException;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.security.spi.authorization.AuthorizationContext;
+import co.cask.cdap.security.spi.authorization.TransactionalFunctionExecutor;
+import co.cask.cdap.security.spi.authorization.TxCallable;
+import co.cask.tephra.TransactionFailureException;
+
+import java.util.Map;
+
+/**
+ * An {@link AuthorizationContext} that delegates to the provided {@link DatasetContext}, {@link Admin} and
+ * {@link Transactional}.
+ */
+public class ForwardingAuthorizationContext implements AuthorizationContext {
+  private final DatasetContext delegateDatasetContext;
+  private final Admin delegateAdmin;
+  private final TransactionalFunctionExecutor rtxnl;
```

Comment:
Good catch - I had it because I had initially called the class ``ReturningTransactional``. I thought that was a weird name, so changed it but forgot to change this var name.